### ISSUE TITLE
High cpu usage after ReadJournal instantiation 

### ISSUE THREAD

**Issue by rodrigotn** [scullxbones/akka-persistence-mongo#185]
Hi, when I just instantiate a ReadJournal something happen with mongo drive (reactivemongo) and cpu usage increase a lot. This is what I'm doing:
```scala
  val readJournal: ScalaDslMongoReadJournal = PersistenceQuery(context.system)
    .readJournalFor[ScalaDslMongoReadJournal](MongoReadJournal.Identifier)
```
Any advice?

Thanks for your help!

Regards,
Rodrigo.

**Comment by scullxbones** [comment]
Hi @rodrigotn -

I'm not sure what would cause that, it's not something I've seen yet.  In production I use the casbah driver, and rely on readJournals heavily - so there may be something either in the integration with rxm or in the rxm driver itself that causes this?  I'd suspect it relates to the tailing/awaiting cursor implementation, but it'd be hard to say for sure without more information.  

Is this against a new database, do the journals have any events in them?

Can you hook up some kind of profiler - VisualVM, or mission control, or whatever you have access to - to see what the hot threads are and pull some stack traces from them?

**Comment by rodrigotn** [comment]
Hi, I have a Thread profiler from newrelic (dont how to export). I tested with an empty database and is the same thing.
![image](https://user-images.githubusercontent.com/28302199/38164260-bc8c0fc0-34d7-11e8-92a6-eaef482d93a7.png)
Thanks!

**Comment by scullxbones** [comment]
@rodrigotn - thanks for the trace, that was the exact type of thing I was looking for.

Unfortunately we're no closer to tying high cpu usage to the plugin or rxm, as these traces do not wind through either the plugin or rxm codebases.  You may need to tweak the settings on New Relic (which unfortunately i've never used) in order to look for traces with the package `akka.contrib.persistence.*` or `reactivemongo.*`.

**Comment by rodrigotn** [comment]
@scullxbones is not possible to do that in newrelic.... but I expanded all entries and I copied:
[rxmongo-profile.txt](https://gist.githubusercontent.com/rodrigotn/1947afa79e1b8aea2bda3b5081fb4b19/raw/07f05e788c749321cbe0d868f05090e57279d06c/rxmongo-profiling.txt)
Let me know if this is useful.

Thanks!

**Comment by scullxbones** [comment]
Hi @rodrigotn -

I took a look through that text file, I'm just not seeing anything referring to the read journal implementation.  One thing you could try is using the casbah driver and not changing anything else to see if that gets rid of the CPU spin.  If the spin goes away, that at least narrows the problem down.

**Comment by rodrigotn** [comment]
Hi @scullxbones , I found the problem: [RxMongoReadJournaller.scala:174](https://github.com/scullxbones/akka-persistence-mongo/blob/master/rxmongo/src/main/scala/akka/contrib/persistence/mongodb/RxMongoReadJournaller.scala#L174). I have commented out this variable and now there is no high cpu usage.
I run all test and one is failing:
![image](https://user-images.githubusercontent.com/28302199/38177910-51de7fc0-35de-11e8-9270-3fbbf19d9ce2.png)

So, do you want that I make a pull request? Or want to analyze the problem?

In the meantime I will work with my compiled version...

Thanks!

**Comment by scullxbones** [comment]
The test failure is spurious, that's unrelated code.

So the problem with commenting that out is that all "live" read streams will be lost.  That will not be a great solution.  I think the issue must be with an empty database and the tailable cursor that this line creates.  I've seen as much elsewhere with tailing cursors in mongo, the starting problem is messy.  If you use a persistent actor to save events to the journal, does the problem also go away?  Or is it still there?

**Comment by rodrigotn** [comment]
I see. Yes, the problem persists with persisted events. Question: the "real time" collection is used by this "live" stream? I have to disable because I'm getting duplicated event using #currentEventByTag(). So, I'm pulling events by hand. Makes sense for you?

**Comment by scullxbones** [comment]
Hi - yes the real time collection is used by the live stream, it's primarily a bus that provides for distributed event queries.  That said, the `current...` series of queries doesn't even use the realtime path, it shouldn't be impacted by disabling the live stream.

I don't think you should be getting duplicated events by tag, if you mean `EventEnvelope`s with duplicate pairs of persistenceId and sequenceNumber.  That would be a bug.

What is meant by "pulling events by hand"?

**Comment by rodrigotn** [comment]
Yes, when I disabled the realtieme persistence stop getting duplicated
events.

I mean that I build my own Graph to simulate a live stream using
currentByTag...

**Comment by scullxbones** [comment]
closed by #187 - released with `2.0.8`.  Thanks @rodrigotn !
